Update changelog with buffer period min behavior change (#654)
hashicorp/consul-terraform-sync: CHANGELOG.md

* Support for Terraform v1.1 [[GH-588](https://github.com/hashicorp/consul-terraform-sync/pull/588)]

IMPROVEMENTS:

> * Library enhancement improves the handling of the min buffer period configuration to more closely match documentation and intended behavior. Tasks will now trigger only after the min time of the buffer period has transpired, previously tasks would trigger immediately, once, before honoring the min buffer period time 

Reviewer: ```suggestion
* Tasks will now always trigger only after the minimum time of the buffer period has transpired. Previously tasks would trigger immediately, once, before honoring the min buffer period configuration.
```
I was thinking that we should lead with the change so it's an easier at-a-glance read. And then I would personally drop the library enhancement and add these details to the PR and link the PR.

Reviewer: Looks good to me!
